static/gsApp/components/profiling/alerts.tsx

```diff
@@ -378,6 +378,169 @@ export function ContinuousProfilingBetaSDKAlertBanner() {
   );
 }
 
+interface ContinuousProfilingNoQuotaAlertBannerProps {
+  organization: Organization;
+  subscription: Subscription;
+}
+
+function ContinuousProfilingNoQuotaAlertAm2Banner({
+  organization,
+  subscription,
+}: ContinuousProfilingNoQuotaAlertBannerProps) {
+  const hasBudget = hasBudgetForContinuousOrUiProfilesHours(subscription);
+
+  // As long as they have a budget set, don't show this banner.
+  if (hasBudget) {
+    return null;
+  }
+
+  // For am2 plans, it's possible they're just using transaction profiles which uses performance units
+  // in this case, we don't want to ask them to set up on-demand budgets for no reason.
+  // So check to see if they're trying to send continuous/ui profiles before shwoing them this banner.
+  if (
+    !subscription.categories.profileDuration?.usageExceeded &&
+    !subscription.categories.profileDurationUI?.usageExceeded
+  ) {
+    return null;
+  }
+
+  const eventTypes: EventType[] = [
+    DataCategoryExact.PROFILE_DURATION,
+    DataCategoryExact.PROFILE_DURATION_UI,
+  ];
+
+  return (
+    <Alert.Container>
+      <Alert
+        system
+        type="warning"
+        showIcon
+        trailingItems={
+          <AddEventsCTA
+            organization={organization}
+            subscription={subscription}
+            buttonProps={{
+              priority: 'default',
+              size: 'xs',
+              style: {marginBlock: `-${space(0.25)}`},
+            }}
+            eventTypes={eventTypes}
+            notificationType="overage_critical"
+            referrer={`no-profiling-quota-${eventTypes.join('-')}`}
+            source="continuous-profiling-no-quota-banner"
+          />
+        }
+      >
+        {t(
+          'Continuous Profiling and UI Profiling requires you to set your On-demand budget.'
+        )}
```

Comment:
```suggestion
        {tct(
          'Continuous Profiling and UI Profiling requires you to set your [budgetTerm] budget.',
          {budgetTerm: subscription.planDetails.budgetTerm}
        )}
```
if you use this looks like you won't need to split into the different tier banners? not sure if there were any other changes

Reply:
im tempted to say let's leave it as separate components as this feels like something that we're likely to iterate on a few times before getting it right and the differences between AM2 and AM3 plans makes it annoying to handle if we have to litter `isAm2Plan` and `isAm3Plan` checks everywhere.